pulp_smash/tests/rpm/cli/test_upload.py

```diff
@@ -0,0 +1,105 @@
+# coding=utf-8
+"""Tests upload of units via client."""
+from __future__ import unicode_literals
+
+import os
+
+import unittest2
+
+from pulp_smash import cli, config, utils
+from pulp_smash.constants import DRPM_URL, DRPM
+from pulp_smash.tests.rpm.utils import set_up_module
+
+
+def setUpModule():  # pylint:disable=invalid-name
+    """Execute ``pulp-admin login`` on the target Pulp system."""
+    set_up_module()
+    utils.pulp_admin_login(config.get_config())
+
+
+class UploadDrpmTestCase(unittest2.TestCase):
+    """Test whether one can upload DRPMs."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a repository and download temporary drpm."""
+        super(UploadDrpmTestCase, cls).setUpClass()
+
+        cls.cfg = config.get_config()
+        cls.repo_id = utils.uuid4()
+        cli.Client(cls.cfg).run(
```

Comment:
Instead of instantiating a cli.Client every time, create a class client and then reuse it for every run call.